Source/Core/Core/Debugger/PPCDebugInterface.cpp

```diff
@@ -383,10 +384,36 @@ void PPCDebugInterface::RunToBreakpoint()
 {
 }
 
+std::shared_ptr<Core::NetworkCaptureLogger> PPCDebugInterface::NetworkLogger()
+{
+  const bool has_ssl = Config::Get(Config::MAIN_NETWORK_SSL_DUMP_READ) ||
+                       Config::Get(Config::MAIN_NETWORK_SSL_DUMP_WRITE);
+  const auto current_capture_type =
+      has_ssl ? Core::NetworkCaptureType::Raw : Core::NetworkCaptureType::None;
+
+  if (m_network_logger && m_network_logger->GetCaptureType() != current_capture_type)
+    m_network_logger.reset();
+
+  if (!m_network_logger)
+  {
+    switch (current_capture_type)
+    {
+    case Core::NetworkCaptureType::Raw:
+      m_network_logger = std::make_shared<Core::BinarySSLCaptureLogger>();
+      break;
+    case Core::NetworkCaptureType::None:
+      m_network_logger = std::make_shared<Core::DummyNetworkCaptureLogger>();
+      break;
+    }
+  }
+  return m_network_logger;
+}
```

Comment:
How do you feel about something along these lines?

```
namespace {
std::shared_ptr<Core::NetworkCaptureLogger> CreateNetworkCaptureLogger(const Core::NetworkCaptureType capture_type) {
	switch(capture_type) {
	case Core::NetworkCaptureType::Raw:
		return std::make_shared<Core::BinarySSLCaptureLogger>();
	case Core::NetworkCaptureType::None:
		return std::make_shared<Core::DummyNetworkCaptureLogger>();	
	}
}
}  // namespace

bool PPCDebugInterface::LoggerHasCaptureType(const Core::NetworkCaptureType capture_type) const {
	return m_network_logger && m_network_logger->GetCaptureType() == capture_type;
}

std::shared_ptr<Core::NetworkCaptureLogger> PPCDebugInterface::NetworkLogger()
{
	const bool has_ssl = Config::Get(Config::MAIN_NETWORK_SSL_DUMP_READ) ||
		Config::Get(Config::MAIN_NETWORK_SSL_DUMP_WRITE);
	const auto current_capture_type = has_ssl ? Core::NetworkCaptureType::Raw : Core::NetworkCaptureType::None;

	if (!LoggerHasCaptureType(current_capture_type))
	{
		m_network_logger = CreateNetworkCaptureLogger(current_capture_type);
	}
	return m_network_logger;
}
```

(Alternately you could assign m_network_logger directly in CreateNetworkCaptureLogger).

Unless I've misunderstood how shared_ptr works you shouldn't need to call reset() before assigning a new value to it, so we don't need to special case that.

Reply:
@Dentomologist - that's basically what I came up with in my comment.  While I thought about it, I didn't bother with the helper functions.  If the logic will be used in multiple places, it might make sense to use helpers but the `NetworkLogger` function is small enough I don't think it's necessary and actually means you have to look somewhere else to know what is happening.  Still, I'd be fine whichever @sepalani chooses!